Core/HLE/sceNetInet.cpp

```diff
@@ -0,0 +1,906 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License 2.0 for more details.
+
+// A copy of the GPL 2.0 should have been included with the program.
+// If not, see http://www.gnu.org/licenses/
+
+// Official git repository and contact information can be found at
+// https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
+
+#if __linux__ || __APPLE__ || defined(__OpenBSD__)
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#endif
+
+#include "Common/Net/Resolve.h"
+#include "Common/Data/Text/Parsers.h"
+
+#include "Common/Serialize/Serializer.h"
+#include "Common/Serialize/SerializeFuncs.h"
+#include "Core/Config.h"
+#include "Core/HLE/HLE.h"
+#include "Core/HLE/FunctionWrappers.h"
+#include "Core/MIPS/MIPS.h"
+#include "Core/MemMapHelpers.h"
+
+#include "Core/HLE/proAdhoc.h"
+#include "Core/HLE/sceNet.h"
+#include "Core/HLE/sceNetInet.h"
+
+#include <iostream>
+#include <shared_mutex>
+
+#include "Core/HLE/sceNp.h"
+#include "Core/Reporting.h"
+// TODO: move Core/Net
+#include "Core/Net/InetCommon.h"
+#include "Core/Net/SceSocket.h"
+
+#if PPSSPP_PLATFORM(SWITCH) && !defined(INADDR_NONE)
+// Missing toolchain define
+#define INADDR_NONE 0xFFFFFFFF
+#elif PPSSPP_PLATFORM(WINDOWS)
+#pragma comment(lib, "ws2_32.lib")
+#define close closesocket
+#define ERROR_WHEN_NONBLOCKING_CALL_OCCURS WSAEWOULDBLOCK
+using netBufferType = char;
+#else
+#define ERROR_WHEN_NONBLOCKING_CALL_OCCURS EWOULDBLOCK
+#include <ifaddrs.h>
+using netBufferType = void;
+#endif
+
+struct SceTimeval {
+	u32 tv_sec;		/* Seconds.  */
+	u32 tv_usec;	/* Microseconds.  */
+};
+
+class SceFdSetOperations {
+public:
+	typedef long int fdMask;
+	static constexpr int gFdsBitsCount = 8 * static_cast<int>(sizeof(fdMask));
+
+	struct FdSet {
+		fdMask mFdsBits[256 / gFdsBitsCount];
+	};
+
+	static void Set(FdSet *sceFdSetBits, int socket) {
+		sceFdSetBits->mFdsBits[Position(socket)] |= ConvertToMask(socket);
+	}
+
+	static bool IsSet(const FdSet *sceFdSetBits, int socket) {
+		return (sceFdSetBits->mFdsBits[Position(socket)] & ConvertToMask(socket)) != 0;
+	}
+
+	static void Clear(FdSet *sceFdSetBits, int socket) {
+		sceFdSetBits->mFdsBits[Position(socket)] &= ~ConvertToMask(socket);
+	}
+
+	static void Zero(FdSet *sceFdSetBits) {
+		memset(sceFdSetBits->mFdsBits, 0, sizeof(FdSet));
+	}
+
+private:
+	static int Position(const int socket) {
+		return socket / gFdsBitsCount;
+	}
+
+	static int ConvertToMask(const int socket) {
+		return static_cast<fdMask>(1UL << (socket % gFdsBitsCount));
+	}
+};
+
+static int getLastError() {
+#if PPSSPP_PLATFORM(WINDOWS)
+	return WSAGetLastError();
+#else
+	return errno;
+#endif
+}
+
+static int sceNetInetInit() {
+    ERROR_LOG(SCENET, "UNTESTED sceNetInetInit()");
+    return SceNetInet::Init() ? 0 : ERROR_NET_INET_ALREADY_INITIALIZED;
+}
+
+int sceNetInetTerm() {
+    ERROR_LOG(SCENET, "UNTESTED sceNetInetTerm()");
+	SceNetInet::Shutdown();
```

Comment:
TODO: Figure out why CLion felt like using tabs instead of spaces ......

Reply:
Was just gonna comment on that :D